geode-assembly/src/test/java/org/apache/geode/management/internal/cli/shell/GfshExitCodeStatusCommandsDUnitTest.java

```diff
@@ -0,0 +1,396 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import org.apache.geode.internal.AvailablePort;
+import org.apache.geode.internal.ExitCode;
+import org.apache.geode.internal.process.PidFile;
+import org.apache.geode.management.internal.cli.util.CommandStringBuilder;
+import org.apache.geode.management.internal.cli.util.ThreePhraseGenerator;
+import org.apache.geode.test.dunit.rules.gfsh.GfshExecution;
+import org.apache.geode.test.dunit.rules.gfsh.GfshRule;
+import org.apache.geode.test.dunit.rules.gfsh.GfshScript;
+import org.apache.geode.test.junit.categories.AcceptanceTest;
+
+// Originally created in response to GEODE-2971
+
+@Category(AcceptanceTest.class)
+@RunWith(JUnitParamsRunner.class)
+public class GfshExitCodeStatusCommandsDUnitTest {
+  private static File toolsJar;
+  private final static ThreePhraseGenerator nameGenerator = new ThreePhraseGenerator();
+  private final static String memberControllerName = "member-controller";
+
+  @Rule
+  public GfshRule gfsh = new GfshRule();
+  private String locatorName;
+  private String serverName;
+
+  private int locatorPort;
+
+  // Some test configuration shorthands
+  private TestConfiguration LOCATOR_ONLINE_BUT_NOT_CONNECTED =
+      new TestConfiguration(true, false, false);
+  private TestConfiguration LOCATOR_ONLINE_AND_CONNECTED = new TestConfiguration(true, false, true);
+  private TestConfiguration BOTH_ONLINE_BUT_NOT_CONNECTED =
+      new TestConfiguration(true, true, false);
+  private TestConfiguration BOTH_ONLINE_AND_CONNECTED = new TestConfiguration(true, true, true);
+
+  @BeforeClass
+  public static void classSetup() {
+    File javaHome = new File(System.getProperty("java.home"));
+    String toolsPath =
+        javaHome.getName().equalsIgnoreCase("jre") ? "../lib/tools.jar" : "lib/tools.jar";
+    toolsJar = new File(javaHome, toolsPath);
+  }
+
+  @Before
+  public void setup() {
+    locatorName = "locator-" + nameGenerator.generate('-');
+    serverName = "server-" + nameGenerator.generate('-');
+    locatorPort = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+  }
+
+  @Test
+  @Parameters(
+      value = {"status locator --port=-10", "status locator --pid=-1", "status server --pid=-1"})
+  public void statusCommandWithInvalidOptionValueShouldFail(String cmd) {
+    GfshScript.of(cmd).withName("test-frame").awaitAtMost(1, MINUTES)
+        .expectExitCode(ExitCode.FATAL.getValue()).execute(gfsh);
+  }
+
+
+  @Test
+  @Parameters(value = {"status locator --host=somehostname", "status locator --port=10334",
+      "status locator --dir=.", "status server --dir=.", "status locator --name=some-locator-name",
+      "status server --name=some-server-name", "status locator --pid=-1", "status server --pid=-1"})
+  public void statusCommandWithValidOptionValueShouldFailWithNoMembers(String cmd) {
```

Comment:
this test failed for me when I run it locally:

[INFO  10:33:30.713 PDT] (test-frame): (1) Executing - status server --dir=.
[INFO  10:33:30.713 PDT] (test-frame): 
[INFO  10:33:30.743 PDT] (test-frame): Server in /private/var/folders/1r/mzzwlxp953ldf540k1cjpyvm0000gn/T/junit5151700688125605579/test-frame on null is currently not responding.
[INFO  10:33:30.743 PDT] (test-frame): 

org.junit.ComparisonFailure: 
Expected :1
Actual   :0
 <Click to see difference>